### PR TITLE
Added sort validation

### LINKED ISSUE #739: Add sort validation to JdbcTaskExecutionDao
Currently the value passed into the `JdbcTaskExecutionDao` via a `Pageable` is not validated. This issue is to add validation to that parameter.

### PR DESCRIPTION
**Pull request by mminella**
This commit now validates that the value passed via a PageRequest to
sort the results by is a valid value.

Resolves #739

### CODE REVIEW

File: spring-cloud-task-core/src/main/java/org/springframework/cloud/task/repository/dao/JdbcTaskExecutionDao.java [hunk 1]
```diff
@@ -161,6 +163,21 @@
 
 	private DataFieldMaxValueIncrementer taskIncrementer;
 
+	private static final Set<String> validSortColumns = new HashSet<>(10);
+
+	static {
+		validSortColumns.add("TASK_EXECUTION_ID");
```
Comment (cppwfs) [inline spring-cloud-task-core/src/main/java/org/springframework/cloud/task/repository/dao/JdbcTaskExecutionDao.java:169]: Do we want these checks to be case insenstive.   i.e. lowercase as well as uppercase.

Comment (mminella) [reply spring-cloud-task-core/src/main/java/org/springframework/cloud/task/repository/dao/JdbcTaskExecutionDao.java:169]: Good catch...updated.

**Comment by cppwfs** [comment]
LGTM

**Comment by cppwfs** [comment]
Rebased Squashed merged

**Pull request closed without merging**